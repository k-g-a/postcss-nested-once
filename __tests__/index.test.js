const fs = require("fs");
const path = require("path");
const { test, expect } = require("@jest/globals");
const postcss = require("postcss");
const nestedOnce = require("../index");
const localByDefault = require("postcss-modules-local-by-default");
const modulesScope = require("postcss-modules-scope");

test("Generates proper output for icss-utils", async () => {
  const from = path.join(__dirname, "__data__", "test.pcss");
  const contents = fs.readFileSync(from);

  const postcssInstance = postcss([
    nestedOnce({ preserveEmpty: true }),
    localByDefault({ mode: "local" }),
    modulesScope({
      generateScopedName: (local, file) => {
        const name = path.basename(file, ".pcss");
        return `${name}_${local}_HASH`;
      },
    }),
  ]);

  const result = await postcssInstance.process(contents, { from: from });

  expect(result.css).toMatchSnapshot("test-icss-compatible-output");
});
