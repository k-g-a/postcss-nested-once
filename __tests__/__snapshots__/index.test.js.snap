// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generates proper output for icss-utils: test-icss-compatible-output 1`] = `
".test_parent_HASH {
  color: red
}

  .test_parent_HASH .test_child_HASH {
    color: green;
  }

.test_list_HASH {
  color: red
}

.test_list_item_HASH {
    color: green;
  }

:export {
  parent: test_parent_HASH;
  child: test_child_HASH;
  list: test_list_HASH;
  list_item: test_list_item_HASH
}
"
`;
